File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} list1
 * @param {ListNode} list2
 * @return {ListNode}
 */
var mergeTwoLists = function(list1, list2) {
    let list3 = new ListNode();
    let c = list3;

    while (list1 && list2){
        if (list1.val > list2.val){
            c.next = list2;
            list2 = list2.next;
        } else {
            c.next = list1;
            list1 = list1.next;
        }
        c = c.next;
    }

    c.next = list1 || list2;

    return list3.next;
};

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode main = new ListNode();
        ListNode cur = main;

        while(list1 != null && list2 != null){
            if (list1.val > list2.val){
                cur.next = list2;
                list2 = list2.next;
            } else {
                cur.next = list1;
                list1 = list1.next;
            }
            // System.out.println(cur.val);
            cur = cur.next;
        }

                
        if (list1 == null){
            cur.next = list2;
        } else {
            cur.next = list1;
        }
        return main.next;
    }
}